proto/redpanda/api/dataplane/v1/mcp.proto

```diff
@@ -0,0 +1,664 @@
+  ];
+
+  // The number of resources that are guaranteed to be assigned to the MCP server.
+  Resources resources = 5 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "The requested amount of resources for the MCP server. Depending on the backend, the resources might be guaranteed, or might be tweaked based on the utilization of the MCP server."}];
+
+  // The current MCP server state.
+  State state = 6 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  Status status = 7 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // URL to connect to the MCP server
+  string url = 8 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // Tags are key-value pairs that can be assigned to a MCP server resource.
+  // They help organize MCP servers and enable filtering when listing them.
+  map<string, string> tags = 9 [(buf.validate.field).map = {
+    max_pairs: 16
+    values: {
+      string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+    }
+  }];
+
+  message Resources {
+    // `memory_shares` is a string specifying the amount of memory to allocate for
+    // the MCP server.
+    //
+    // This follows the [Kubernetes quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) format. Acceptable units
+    // include:
+    // - Decimal SI units: "K", "M", "G", "T", "P", "E" (e.g., "128M" for 128
+    //   megabytes)
+    // - Binary SI units: "Ki", "Mi", "Gi", "Ti", "Pi", "Ei" (e.g., "512Mi" for
+    // 512 mebibytes) If you don't specify a unit, the value is interpreted as
+    // bytes.
+    string memory_shares = 1 [
+      (google.api.field_behavior) = REQUIRED,
+      (buf.validate.field).required = true
+    ];
+    // `cpu_shares` is a string specifying the amount of CPU to allocate for the
+    // MCP server.
+    //
+    // This follows the [Kubernetes quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) format. Acceptable
+    // units include:
+    // - Decimal SI units: "m" (e.g., "500m" for 500 millicores, "2" for 2 cores)
+    // CPU shares can be specified in millicores (1 core = 1000 millicores).
+    // If you don't specify a unit, the value is interpreted as the number of cores.
+    string cpu_shares = 2 [
+      (google.api.field_behavior) = REQUIRED,
+      (buf.validate.field).required = true
+    ];
+  }
+
+  // State of the MCP server.
+  enum State {
+    STATE_UNSPECIFIED = 0;
+    // The MCP server is starting.
+    STATE_STARTING = 1;
+    // The MCP server is running.
+    STATE_RUNNING = 2;
+    // The MCP server is in the process of stopping.
+    STATE_STOPPING = 3;
+    // The MCP server is stopped and in paused state.
+    STATE_STOPPED = 4;
+    // The MCP server encountered an error.
+    STATE_ERROR = 5;
+    // The MCP server has completed the job successfully.
+    STATE_COMPLETED = 6;
```

Comment:
i think completed makes no sense for MCP. it's only for oneshot pipelines, but mcp is never oneshot ("server")

Reply:
done